ParameterCalculator.py

```diff
@@ -0,0 +1,198 @@
+    attackFrequencies, sustainFrequencies, decayFrequencies, fullFrequencies = 0, 0, 0, 0
+    for take in takes:
+        fullFrequencies, fullSpectrum = signal.periodogram(take, samplingRate, scaling="spectrum")
+        attackFrequencies, attackSpectrum = signal.periodogram(take[:int(attackTime*samplingRate)], samplingRate, scaling="spectrum")
+        sustainFrequencies, sustainSpectrum = signal.periodogram(take[int(attackTime*samplingRate):int(sustainTime*samplingRate)], samplingRate, scaling="spectrum")
+        decayFrequencies, decaySpectrum = signal.periodogram(take[int(sustainTime*samplingRate):], samplingRate, scaling="spectrum")
+
+        fullSpectrums = InsertIntoVstack(fullSpectrum.real, fullSpectrums)
+        attackSpectrums = InsertIntoVstack(attackSpectrum.real, attackSpectrums)
+        sustainSpectrums = InsertIntoVstack(sustainSpectrum.real, sustainSpectrums)
+        decaySpectrums = InsertIntoVstack(decaySpectrum.real, decaySpectrums)
+
+    return fullSpectrums, fullFrequencies, attackFrequencies, attackSpectrums, sustainFrequencies, sustainSpectrums, decayFrequencies, decaySpectrums
+
+# Creates vector of mathematical harmonic frequencies based on the fundamental pitch
+def CreateMathematicalHarmonicFrequencyVector(pitch, n):
+    freq = []
+    for i in range (1, n):
+        freq.append(pitch * i)
+    return freq
+
+# Based on harmonic data from iracema estimates the fundamental pitch
+def EstimateFundamentalPitch(harmonicFrequencies):
+    fundamentalPitches = []
+    for i in range (0, len(harmonicFrequencies)):
+        fundamentalPitches.append(harmonicFrequencies[i]/(i+1))
+    return np.mean(fundamentalPitches)
+
+# Takes spectrum of impulse and the mathematical harmonic frequencies and returns the amplitudes and frequencies of harmonics in the signal
+def ExtractHarmonicDataFromSpectrums(spectrums, spectrumFrequencies, mathHarmonicFrequencies, bufforInHZ = 5):
+    buffor = int(bufforInHZ/(spectrumFrequencies[1]-spectrumFrequencies[0]))
+    harmonicData = []
+
+    for spectrum in spectrums:
+        amplitudes = []
+        harmonicFrequencies = []
+        harmonicNumber = 0
+
+        for fftSample in range (0, len(spectrum)):
+            if spectrumFrequencies[fftSample] > mathHarmonicFrequencies[harmonicNumber]:
+                peakValue = 0
+                peakFrequency = 0
+
+                for i in range (fftSample-buffor, fftSample+buffor):
+                    if spectrum[i] > peakValue:
+                        peakValue = spectrum[i]
+                        peakFrequency = spectrumFrequencies[i]
+
+                amplitudes.append(peakValue)
+                harmonicFrequencies.append(peakFrequency)
+
+                if harmonicNumber < len(mathHarmonicFrequencies) - 1:
+                    harmonicNumber = harmonicNumber + 1
+                else:
+                    break
+
+        # showing found harmonics and spectrum for debugging
+        if False:
```

Comment:
I mean, xD weź daj tu jakąś stałą debugową żeby wystarczyło ją ustawić w jednym miejscu

Reply:
Aaa... o tym zapominełem -.-